src/pile.js

```diff
@@ -1032,6 +1032,54 @@ const createPile = (
     setCover(newCover);
   };
 
+  let labelGraphics;
+  let labelTextContainer;
+
+  const drawLabel = (labels, colors, texts) => {
+    if (isPositioning || isScaling) {
+      postPilePositionAnimation.set('drawLabel', () => {
+        drawLabel(labels, colors, texts);
+      });
+      return;
+    }
+
+    if (!labelGraphics || !labelTextContainer) {
+      labelGraphics = new PIXI.Graphics();
+      labelTextContainer = new PIXI.Container();
+      contentGraphics.addChild(labelGraphics);
+      contentGraphics.addChild(labelTextContainer);
+    } else {
+      labelGraphics.clear();
+      labelTextContainer.removeChildren();
+    }
+
+    const { width, height } = contentGraphics;
+
+    if (labels.length) {
+      const labelWidth = width / labels.length;
+      const labelHeight = 15;
+      labels.forEach((label, index) => {
+        const labelX = labelWidth * index - width / 2;
+        const color = colors[index];
+        labelGraphics.beginFill(color, 1);
+        labelGraphics.drawRect(labelX, height / 2, labelWidth, labelHeight);
+        labelGraphics.endFill();
+      });
+    }
+
+    if (texts.length) {
+      const textWidth = width / texts.length;
+      texts.forEach((text, index) => {
+        const labelText = new PIXI.Text(text);
+        labelText.anchor.set(0.5, 0);
+        labelText.x = textWidth * index - width / 2 + textWidth / 2;
+        labelText.y = height / 2;
+        labelText.style.fontSize = 15;
```

Comment:
The font-size should ideally be user-controllable as well but at least it should be smaller by default :) Maybe 8px?